datafusion/sql/src/unparser/expr.rs

```diff
@@ -514,6 +516,57 @@ impl Unparser<'_> {
         })
     }
 
+    fn named_struct_to_sql(&self, args: &[Expr]) -> Result<ast::Expr> {
+        if args.len() % 2 != 0 {
+            return internal_err!("named_struct must have an even number of arguments");
+        }
+
+        let args = args
+            .chunks_exact(2)
+            .map(|chunk| {
+                let key = match &chunk[0] {
+                    Expr::Literal(lit) => self.new_ident_quoted_if_needs(lit.to_string()),
```

Comment:
```suggestion
                    Expr::Literal(ScalarValue::Utf8(Some(s))) => self.new_ident_quoted_if_needs(s.to_string()),
```
Refer to the implementation of `named_struct`, we can match the string literal only.
https://github.com/apache/datafusion/blob/9fb7aee95c5fcf177609963cedadf443ba6fe1b7/datafusion/functions/src/core/named_struct.rs#L49-L52